accepted/00000-remote-command-auditing.md

```diff
@@ -0,0 +1,106 @@
+
+- Role-based restriction for `Remote Command`: a new role will be introduced. Every user associated with this role (in addition to either `SUSE Manager Administrator` or `Organization Administrator`) can access the `Remote Command` feature and issue commands. Every other user will not see the `Remote Command` tab nor the `Salt > Remote Commands` menu item.
+- Auditing for `Remote Command` and `Salt > Remote Commands`: the product will log the user that has issued the command, the command itself and its result.
+
+# Detailed design
+[design]: #detailed-design
+
+The first step is to introduce a new role in the roles list: `Remote Command Administrator` (`RhnUserGroupType` table).
+The role is a normal role (comparable to `Config Administrator`, `Image Administrator`, etc) and not an `Administrative Role` (`SUSE Manager Administrator` or `Organization Administrator`). If a user has a `Remote Command Administrator` role but not one of the `Administrative Role`s, then the user cannot see any system and thus should behave like the other normal roles.
+
+In the Java backend, the following menu items and associated pages must be hidden and not accessible if the logged in user does not have the `Remote Command Administrator` role + an `Administrative Role` (`UserImpl.hasPermanentRole`) or there are no systems manageable by the user (count needs to be cached):
+
+- Traditional systems and Salt systems: `path="/systems/details/SystemRemoteCommand"`, `path="/systems/ssm/provisioning/RemoteCommand"` and `SystemRemoteCommandAction` and `ProvisioningRemoteCommand` Java classes.
+- Salt systems: `path="/manager/systems/cmd"` and `RemoteMinionCommands` Java class.
+
+NOTE: the API call and the `spacecmd` command (`system_runscript`) will still be visible independent of the role.
+
+Additionally, the two mentioned Java classes must implement auditing in the form of logging.
+A new file that will be placed under `/var/log/rhn/` and it will be called `rhn_remote_commands_audit.log` will contain entries in the form:
+
+```
+[timestamp in UTC]
+command executed by the user
+target: 
+  - "local" + system if RemoteCommand on a single system or SSM 
+  - "target-pattern" + pattern if RemoteMinionCommands action
+result
+login of the user in the product that has issued the command
+the IP address of the user originating the command
+```
+
+Example:
+
+```
+[21/Jun/2019:10:57:50 +0000]
+"ls -l"
+"local" + suma-refhead-min-centos7.mgr.suse.de
+"total 0\n-rw-r--r-- 1 mbologna users 0 Jun 21 13:09 test"
+mbologna
+10.31.33.7
+
+[22/Jun/2019:11:32:50 +0000]
+"rm -fr /"
+"target-pattern" + '*'
+""
+nastyuser
+192.168.18.81
+```
+
+The final format of the log could be formatted in a JSON fashion to be easily parseable.
+The log file should be rotated and archived as usual product logs.
+
+The `Remote Command Administrator` role must be explicitly set. By default, the `Remote Command Administrator` role is not assigned to a new user.
```

Comment:
Current behavior of all roles is:

- nothing is selected by default
- when you select "Org Admin" all Roles are auto-selected.
- Same for "SUSE Manager Admin"

We should not break this concept. It maybe deeply integrated in the code.